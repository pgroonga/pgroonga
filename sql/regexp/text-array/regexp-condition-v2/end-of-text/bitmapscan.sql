CREATE TABLE memos (
  contents text[]
);

INSERT INTO memos
     VALUES (ARRAY['PostgreSQL is a RDBMS',
                   'Groonga is fast full text search engine',
                   'PGroonga is a PostgreSQL extension that uses Groonga']);

INSERT INTO memos
     VALUES (ARRAY['MySQL is a RDBMS',
                   'Mroonga is a MySQL storage engine that uses Groonga']);

CREATE INDEX pgrn_contents_index
    ON memos
 USING pgroonga (contents pgroonga_text_array_regexp_ops_v2);

SET enable_seqscan = off;
SET enable_indexscan = off;
SET enable_bitmapscan = on;

EXPLAIN (COSTS OFF)
SELECT *
  FROM memos
 WHERE contents &~ pgroonga_condition('engine\z');

SELECT *
  FROM memos
 WHERE contents &~ pgroonga_condition('engine\z');

DROP TABLE memos;
