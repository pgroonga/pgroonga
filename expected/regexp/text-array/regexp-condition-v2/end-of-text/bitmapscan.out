CREATE TABLE memos (
  contents text[]
);
INSERT INTO memos
     VALUES (ARRAY['PostgreSQL is a RDBMS',
                   'Groonga is fast full text search engine',
                   'PGroonga is a PostgreSQL extension that uses Groonga']);
INSERT INTO memos
     VALUES (ARRAY['MySQL is a RDBMS',
                   'Mroonga is a MySQL storage engine that uses Groonga']);
CREATE INDEX pgrn_contents_index
    ON memos
 USING pgroonga (contents pgroonga_text_array_regexp_ops_v2);
SET enable_seqscan = off;
SET enable_indexscan = off;
SET enable_bitmapscan = on;
EXPLAIN (COSTS OFF)
SELECT *
  FROM memos
 WHERE contents &~ pgroonga_condition('engine\z',
                                      index_name => 'pgrn_contents_index');
                                           QUERY PLAN                                           
------------------------------------------------------------------------------------------------
 Bitmap Heap Scan on memos
   Recheck Cond: (contents &~ '("engine\\z",,,,pgrn_contents_index,,)'::pgroonga_condition)
   ->  Bitmap Index Scan on pgrn_contents_index
         Index Cond: (contents &~ '("engine\\z",,,,pgrn_contents_index,,)'::pgroonga_condition)
(4 rows)

SELECT *
  FROM memos
 WHERE contents &~ pgroonga_condition('engine\z',
                                      index_name => 'pgrn_contents_index');
                                                          contents                                                          
----------------------------------------------------------------------------------------------------------------------------
 {"PostgreSQL is a RDBMS","Groonga is fast full text search engine","PGroonga is a PostgreSQL extension that uses Groonga"}
(1 row)

DROP TABLE memos;
